code/game/machinery/_machinery.dm

```diff
@@ -248,24 +248,39 @@ Class Procs:
 	return !(machine_stat & (NOPOWER|BROKEN|MAINT))
 
 /obj/machinery/can_interact(mob/user)
-	var/silicon = issiliconoradminghost(user)
-	if((machine_stat & (NOPOWER|BROKEN)) && !(interaction_flags_machine & INTERACT_MACHINE_OFFLINE))
+	var/silicon = issilicon(user)
+	var/admin_ghost = IsAdminGhost(user)
+	var/living = isliving(user)
```

Comment:
can you add these further down when they're actually needed to avoid calling all 3 procs even if the machine is offline / broken / etc

vector for abuse